src/edge/tests/test_ssr.py

```diff
@@ -853,6 +853,171 @@ def allowed():
         )
 
 
+class InversionRecombinationAnnotationTest(TestCase):
+
+    def test_can_invert_using_same_site_with_annotations(self):
+        parent_genome = Genome(name="foo")
+        parent_genome.save()
+        f = Fragment.create_with_sequence(
+            "bar",
+            "attg" + "atg" + "caat" + "c" * 1000
+        )
+        f.annotate(6, 7, "flipped feature", "gene", 1)
+        f.annotate(3, 9, "truncated flipped feature", "gene", -1)
+
+        Genome_Fragment(genome=parent_genome, fragment=f, inherited=False).save()
+
+        class FakeReaction(Reaction):
+            @staticmethod
+            def allowed():
+                return [Inversion("attg", "caat", "attg", "caat")]
+
+        r = FakeReaction(parent_genome, None, None)
+        child_genome = r.run_reaction("far")
+        f = child_genome.fragments.all()[0].indexed_fragment()
+        self.assertEquals(f.sequence, "attg" + "cat" + "caat" + "c" * 1000)
+
+        anns = f.annotations()
+        self.assertEqual(len(anns), 2)
+
+        self.assertEqual(anns[0].base_first, 5)
```

Comment:
Should we check feature_base_first and such, to make sure the annotation represents a partial annotation? Also, would we want to preserve the annotation at bps 3-4, and 8-9? Those are from the original annotation.

Reply:
I'm open to alternate spec, but currently, the annotation logic will just generate a completely new feature on the opposite strand, once the sequence is inverted. Even if partial, we don't reuse or associate the new annotation to the old one. Then, regardless if the annotation is partially or completely on the sequence to invert, we just grab whatever we can for the new annotation via the following:
`truncated_base_first = max(self.inverted_sequence_start, ann.base_first)
truncated_base_last = min(self.inverted_sequence_end, ann.base_last)`

The annotations aren't preserved at 3-4 or 8-9 as they cover the left and right sites, which might be subject to change if the `recombined_site_left` and/or `recombined_site_right` end up being different for the specified Inversion event (in this case they're just the same), though any sequence conserved outside of those sites will preserve annotations, e.g. if the annotation was 3-19 instead of 3-9, that chunk of 12-19 would be preserved still.